src/sage/groups/perm_gps/partn_ref/data_structures.pyx

```diff
@@ -292,8 +294,7 @@ cdef int PS_first_smallest(PartitionStack *PS, bitset_t b, int *second_pos=NULL,
     bitset_zero(b)
     while 1:
```

Comment:
Isn't it safer to use a `for` loop to avoid exceeding the largest possible value of `I` ?